index.js

```diff
@@ -241,26 +264,26 @@ const ContainerService = {
       return Promise.resolve(null);
     }
 
-    return new Promise(resolve => {
-      this._containerTabIterator(args.userContextId, tab => {
-        const object = this._createTabObject(tab);
+    const tabsToClose = [];
 
-        // This tab is going to be closed. Let's mark this tabObject as
-        // non-active.
-        object.active = false;
+    this._containerTabIterator(args.userContextId, tab => {
+      const object = this._createTabObject(tab);
 
-        getFavicon(object.url).then(url => {
-          object.favicon = url;
-        }).catch(() => {
-          object.favicon = "";
-        });
+      // This tab is going to be closed. Let's mark this tabObject as
+      // non-active.
+      object.active = false;
 
-        this._identitiesState[args.userContextId].hiddenTabUrls.push(object);
-        tab.close();
+      getFavicon(object.url).then(url => {
+        object.favicon = url;
+      }).catch(() => {
+        object.favicon = "";
```

Comment:
Not new behavior in this patch, but why are we assigning the `object.favicon` here before we hide/close the tabs? We already get the favicon in `getTabs` right?